libusb/descriptor.c

```diff
@@ -1517,3 +1517,116 @@ void API_EXPORTED libusb_free_interface_association_descriptors(
 		free((void*)iad_array->iad);
 	free(iad_array);
 }
+
+/*
+ * \brief Copy a UTF-8 string with proper truncation if needed.
+ *
+ * \param tgt The target utf-8 string
+ * \param src The source utf-8 string.
+ * \param tgt_size The size of target in bytes.
+ * \return the length of src in bytes, including the null terminator.
+ *
+ * utf8_copy(NULL, src, 0) is equivalent to strlen(src) + 1.
+ */
+static int usbi_utf8_copy(char* tgt, char const* src, size_t tgt_size) {
```

Comment:
here we have this same `int` vs `size_t` thing... looks like there's only one caller of `usbi_utf8_copy` and it passes an int variable for the last parameter, so I guess you can change it to `int`. Then you wouldn't need those truncating casts either.

Reply:
Sure.  Will change.